tutorials/animation/index.md

```diff
@@ -0,0 +1,874 @@
+    );
+  }
+}
+{% endprettify %}
+
+The middle three blocks in the diagram are all created in the
+`build()` method in GrowTransition. The GrowTransition widget itself
+is stateless and holds the set of final variables necessary to
+define the transition animation. The build() function creates and
+returns the AnimatedBuilder, which takes the (Anonymous builder)
+method and the LogoWidget object as parameters. The work of
+rendering the transition actually happens in the (Anonymous builder)
+method, which creates a Container of the appropriate size to force
+the LogoWidget to shrink to fit.
+
+One tricky point in the code below is that the child looks like it's
+specified twice. What's happening is that the outer reference of
+child is passed to AnimatedBuilder, which passes it to the anonymous
+closure, which then uses that object as its child. The net result is
+that the AnimatedBuilder is inserted in between the two widgets in
+the render tree.
+
+<!-- skip -->
+{% prettify dart %}
+class GrowTransition extends StatelessWidget {
+  GrowTransition({this.child, this.animation});
+
+  final Widget child;
+  final Animation<double> animation;
+
+  @override
+  Widget build(BuildContext context) {
+    return new AnimatedBuilder(animation: animation,
+        builder: (BuildContext context, Widget child) {
+          return new Center(
+              child: new Container(
+                  height: animation.value,
+                  width: animation.value,
+                  child: child));
+        }, child: child);
+  }
+}
+{% endprettify %}
+
+Finally, the code to initialize the animation looks very similar to
+the first example,
+[animate1.](https://raw.githubusercontent.com/InMatrix/animation_tutorial/master/lib/animate1.dart)
+The `initState()` method creates an AnimationController
+and a Tween, then binds them with `animate()`. The magic happens in the
+`build()` methods, which returns a GrowTransition object with a
+LogoWidget as a child, and an animation object to drive the transition.
+These are the three elements listed in the bullet points above.
+
+
+<!-- skip -->
+{% prettify dart %}
+class LogoApp extends StatefulWidget {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> with TickerProviderStateMixin {
+  Animation animation;
+  AnimationController controller;
+
+  initState() {
+    super.initState();
+    controller =
+    new AnimationController(
+        duration: const Duration(milliseconds: 2000),
+        vsync: this,
+    );
+    final curve = new CurvedAnimation(
+        parent: controller,
+        curve: Curves.easeIn
+    );
+    animation = new Tween(begin: 0.0, end: 300.0).animate(curve);
+    controller.forward();
+  }
+
+  Widget build(BuildContext context) {
+    return new GrowTransition(
+        child: new LogoWidget(),
+        animation: animation
+    );
+  }
+
+  @override
+  dispose() {
+    super.dispose();
+    controller.dispose();
```

Comment:
here and elsewhere: convention is to call `super.dispose()` last, since in principle once you've called it your object is dead and you shouldn't do anything anymore.